### PR TITLE
Fix issue #36 and feature toggle for #37

### PR DESCRIPTION
**Pull request by strelok1**
- Fixed the case where there are many "closed" repos and only one open one (note to promote in this case you MUST close and promote together)
- Added feature toggle for "nexus-staging.autoDropAfterRelease" via
project property as per #37
- Fixed typos and tests

### CODE REVIEW

File: src/main/groovy/io/codearte/gradle/nexus/logic/RepositoryFetcher.groovy [hunk 1]
```diff
@@ -24,24 +24,16 @@ class RepositoryFetcher extends BaseOperationExecutor {
...
-        }
-        Map repository = responseAsMap.data[0] as Map
-        return repository
-    }
-
-    private void verifyReceivedRepositoryState(Map repository, String expectedRepositoryState) {
-        if (repository.type != expectedRepositoryState) {
-            throw new IllegalArgumentException(
-                    "Unexpected state of received repository. Received ${repository.type}, expected $expectedRepositoryState")
+        def repositoryInGivenState = { it.type == repositoryState }
```
Comment (szpak) [inline src/main/groovy/io/codearte/gradle/nexus/logic/RepositoryFetcher.groovy:32]: 👍

**Comment by szpak** [comment]
Thanks for your PR.

In addition to checking number of repositories in a given state I like the idea to reause staging repository id. However, `autoDropAfterRelease` is not related to that (you could make at least a separate commit for that). As I described in #37 the change is simply, but it is a not documented feature and I need to verify it before a release. Preferably I would run Nexus in Docker for my acceptance tests, but AFAIK that feature is not available in open source version of Nexus and getting a perpetual trial for Docker Professional can be problematic. I plan to work on your PR a little bit (probably this week) and merge it later on.

Btw, if you need that changes earlier I propose you to use [JitPack](http://jitpack.io/) to build a version from your commit.

**Comment by strelok1** [comment]
Cool, that's why I put the fix for #37 behind a project property. I will take it out of this PR and make a separate commit later today. Cheers.

**Comment by szpak** [comment]
@strelok1 No need to worry. I plan to make it a permanent feature (unless you see a reason why anyone could do not like it). I internally already (partially) reworked your PR, so please do not touch it :). I just need to improve testing infrastructure in general before making bigger changes which I have planned for Q2 :).

**Comment by strelok1** [comment]
Ok no prob. Leaving it with you.

**Pull request merged**